test/mix/tasks/compile/surface_test.exs

```diff
@@ -86,6 +86,61 @@ defmodule Mix.Tasks.Compile.SurfaceTest do
            """
   end
 
+  test "update dest hook file and index.js if there are changes in hooks" do
+    refute File.exists?(@hooks_abs_output_dir)
+
+    index_file = Path.join(@hooks_abs_output_dir, "index.js")
+    src_hooks_file = Path.join(@test_components_dir, "fake_button.hooks.js")
+
+    dest_hooks_file =
+      Path.join(@hooks_abs_output_dir, "Mix.Tasks.Compile.SurfaceTest.FakeButton.hooks.js")
+
+    %File.Stat{mtime: src_hooks_file_time} = File.stat!(src_hooks_file)
+
+    run([])
+
+    new_time = add_secs(src_hooks_file_time, -1)
+    File.touch!(index_file, new_time)
+    File.touch!(dest_hooks_file, new_time)
+    %File.Stat{mtime: index_file_time} = File.stat!(index_file)
+    %File.Stat{mtime: dest_hooks_file_time} = File.stat!(dest_hooks_file)
+
+    assert index_file_time < src_hooks_file_time
+    assert dest_hooks_file_time < src_hooks_file_time
+
+    run([])
+
+    %File.Stat{mtime: index_file_time} = File.stat!(index_file)
+    %File.Stat{mtime: dest_hooks_file_time} = File.stat!(dest_hooks_file)
+
+    assert index_file_time > src_hooks_file_time
+    assert dest_hooks_file_time > src_hooks_file_time
```

Comment:
I don't like to have to touch the files to be able to check if were updated but since `File.stat!` returns the modification time in seconds, I couldn't find a better way.